src/server/JasmineGraphInstanceService.cpp

```diff
@@ -80,4301 +144,5243 @@ void *instanceservicesession(void *dummyPt) {
+    std::string dataFolder = Utils::getJasmineGraphProperty("org.jasminegraph.server.instance.datafolder");
+    map<long, unordered_set<long>> aggregatedCompositeCentralStore;
 
-            write(connFd, JasmineGraphInstanceProtocol::OK.c_str(), JasmineGraphInstanceProtocol::OK.size());
-            instance_logger.log("Sent : " + JasmineGraphInstanceProtocol::OK, "info");
+    std::vector<std::string> compositeCentralStoreFileList = Utils::split(compositeFileList, ':');
+    std::vector<std::string>::iterator compositeCentralStoreFileIterator;
+    std::vector<std::string> availableCompositeFileList = Utils::split(availableFileList, ':');
+    std::vector<std::string>::iterator availableCompositeFileIterator;
 
-            bzero(data, INSTANCE_DATA_LENGTH + 1);
-            read(connFd, data, INSTANCE_DATA_LENGTH);
-            string workerList = (data);
-            workerList = Utils::trim_copy(workerList, " \f\n\r\t\v");
-            instance_logger.log("Received Worker List " + workerList, "info");
-
-            std::vector<string> workerSockets;
-            stringstream wl(workerList);
-            string intermediate;
-            while (getline(wl, intermediate, ',')) {
-                workerSockets.push_back(intermediate);
-            }
+    for (availableCompositeFileIterator = availableCompositeFileList.begin();
+         availableCompositeFileIterator != availableCompositeFileList.end(); ++availableCompositeFileIterator) {
+        std::string availableCompositeFileName = *availableCompositeFileIterator;
+        size_t lastindex = availableCompositeFileName.find_last_of(".");
+        string rawFileName = availableCompositeFileName.substr(0, lastindex);
+        struct stat st;
 
-            // Calculate the in degree distribution
-            map<long, long> degreeDistribution =
-                calculateInDegreeDist(graphID, partitionID, serverPort, graphDBMapLocalStores, graphDBMapCentralStores,
-                                      workerSockets, workerList);
-            degreeDistribution.clear();
-            loop = true;
-        } else if (line.compare(JasmineGraphInstanceProtocol::WORKER_OUT_DEGREE_DISTRIBUTION) == 0) {
-            instance_logger.log("Received : Out degree distribution to aggregator", "info");
+        std::string availableCompositeFile = dataFolder + "/" + rawFileName;
 
-            write(connFd, JasmineGraphInstanceProtocol::OK.c_str(), JasmineGraphInstanceProtocol::OK.size());
-            instance_logger.log("Sent : " + JasmineGraphInstanceProtocol::OK, "info");
-            bzero(data, INSTANCE_DATA_LENGTH + 1);
-            read(connFd, data, INSTANCE_DATA_LENGTH);
-            string graphID = (data);
-            graphID = Utils::trim_copy(graphID, " \f\n\r\t\v");
-            instance_logger.log("Received Graph ID: " + graphID, "info");
+        if (stat(availableCompositeFile.c_str(), &st) == 0) {
+            if (st.st_mode & S_IFREG) {
```

Comment:
Need to check why its bitwise & and not being logical &

Reply:
It's checking if it's a regular file. It should be bit-wise &.

Reply:
Ok